src/executables/csql.c

```diff
@@ -3352,98 +3359,54 @@ get_host_ip (unsigned char *ip_addr)
   return 0;
 }
 
-/*
-int
-csql_arg_copy (CSQL_ARGUMENT * dest, CSQL_ARGUMENT * src)
-{
-
-  dest->sa_mode = src->sa_mode;
-  dest->cs_mode = src->cs_mode;
-  dest->single_line_execution = src->single_line_execution;
-  dest->column_output = src->column_output;
-  dest->line_output = src->line_output;
-  dest->read_only = src->read_only;
-  dest->auto_commit = src->auto_commit;
-  dest->nopager = src->nopager;
-  dest->continue_on_error = src->continue_on_error;
-  dest->sysadm = src->sysadm;
-  dest->write_on_standby = src->write_on_standby;
-  dest->trigger_action_flag = src->trigger_action_flag;
-  dest->plain_output = src->plain_output;
-  dest->skip_column_names = src->skip_column_names;
-  dest->skip_vacuum = src->skip_vacuum;
-  dest->string_width = src->string_width;
-  dest->query_output = src->query_output;
-  dest->column_delimiter = src->column_delimiter;
-  dest->column_enclosure = src->column_enclosure;
-  dest->loaddb_output = src->loaddb_output;
-  dest->no_logging = src->no_logging;
-
-  return 0;
-}
-*/
 static int
 csql_connect (char *argument, CSQL_ARGUMENT * csql_arg)
 {
-  /*dbname_cpy can be stored DB name+ @ + remote hostname + \0 */
-  char dbname_cpy[MAXLOGNAME + CUB_MAXHOSTNAMELEN + 2];
-  char username_cpy[DB_MAX_USER_LENGTH + 1];
-  char host_cpy[DB_MAX_USER_LENGTH + 1];
+  /*dbname can be stored DB name+ @ + remote hostname + \0 */
+  char dbname[MAXLOGNAME + CUB_MAXHOSTNAMELEN + 2];
+  char username[DB_MAX_USER_LENGTH + 1];
   const char *delim = " @\n";
   char *save_ptr_strtok = NULL;
-  char *db_name = NULL;
-  char *user_name = NULL;
-  char *host_name = NULL;
+  char *db_name_ptr = NULL;
+  char *user_name_ptr = NULL;
+  char *host_name_ptr = NULL;
   char *p = NULL;
   int error_code;
   CSQL_ARGUMENT csql_new_arg;
 
-  /*find @hostname */
-  host_name = strchr (argument, '@');
-  if (host_name != NULL)
+  /*verify @hostname */
+  host_name_ptr = strchr (argument, '@');
+  if (host_name_ptr != NULL)
     {
-      if (*(host_name + 1) == '\0')
+      if (*(host_name_ptr + 1) == '\0')
 	{
-	  host_name = NULL;
+	  host_name_ptr = NULL;
 	  csql_Error_code = CSQL_ERR_CONNECT;
 	  return DO_CMD_FAILURE;
 	}
-      else
-	{
-	  host_name += 1;
-	}
-    }
-
-  if (host_name != NULL)
-    {
-      if (strcasecmp ("127.0.0.1", host_name) == 0 || strcasecmp ("localhost", host_name) == 0
-	  || strcasecmp (boot_get_host_name (), host_name) == 0 || strcasecmp (boot_get_ip (), host_name) == 0)
-	{
-	  *(host_name - 1) = '\0';
-	}
     }
 
-  strcpy (dbname_cpy, argument);
-  strcpy (username_cpy, argument);
+  strcpy (dbname, argument);
+  strcpy (username, argument);
 
   /*find db name following the user name */
-  db_name = strrchr (dbname_cpy, ' ');
-  if (db_name != NULL)
+  db_name_ptr = strrchr (dbname, ' ');
+  if (db_name_ptr != NULL)
```

Comment:
How about processing db_name_ptr in one if statement?  

```
  if (db_name_ptr != NULL)
    {
      if (*(db_name_ptr + 1) == '\0')
	{
	  db_name_ptr = NULL;
	  csql_Error_code = CSQL_ERR_CONNECT;
	  return DO_CMD_FAILURE;
	}
      else
	{
	  db_name_ptr += 1;
	}
    }
  else
    {
      db_name_ptr = strdup (csql_arg->db_name);
    }
```

Reply:
Assignment to local variable db_name_ptr to NULL before return is not required.

```
  if (db_name_ptr != NULL)
    {
      if (*(db_name_ptr + 1) == '\0')
	{
	  csql_Error_code = CSQL_ERR_CONNECT;
	  return DO_CMD_FAILURE;
	}
      else
	{
	  db_name_ptr += 1;
	}
    }
  else
    {
      db_name_ptr = strdup (csql_arg->db_name);
    }

```

Reply:
Thank you it will be better.